sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQueryBuilder.cs

```diff
@@ -87,29 +89,67 @@ public FromQuery SelectCount()
         /// .Build();
         /// </code>
         /// </example>
-        /// <param name="customQuery"> Query in string format. </param>
+        /// <param name="customQuery">Query in string format.</param>
         /// <returns> Query that contains a select clause. </returns>
-        public FromQuery SelectCustom(string customQuery)
+        public SelectAsQuery SelectCustom(string customQuery)
         {
             return _selectQuery.SelectCustom(customQuery);
         }
 
+        /// <summary>
+        /// Used to alias selectable properties in place of the AS keyword.
+        /// </summary>
+        /// <param name="field">The proper name for the selectable property in the ADT Query Language.</param>
+        /// <param name="alias">The alias to be assigned to the return contents in the query response.</param>
+        /// <returns> Query that contains an aliased select clause. </returns>
+        public SelectAsQuery SelectAs(string field, string alias)
+        {
+            return _selectAs.SelectAs(field, alias);
+        }
+
         /// <summary>
         /// Gets the string representation of the built query.
         /// </summary>
         /// <returns> String represenation of query. </returns>
         public string GetQueryText()
         {
             var finalQuery = new StringBuilder();
+            finalQuery.Append(QueryConstants.Select).Append(' ');
 
             // build the select string
-            finalQuery.Append(_selectQuery.GetQueryText());
+            string selectStatementQueryText = _selectQuery.GetQueryText();
+            string selectStatementAliasedQueryText = _selectAs.GetQueryText();
+
+            bool nonAliasedSelectStatements = selectStatementQueryText.Length > 0;
+            bool aliasedSelectStatements = selectStatementAliasedQueryText.Length > 0;
+
+            // add aliased select components
+            if (nonAliasedSelectStatements)
+            {
+                // begin with any non-aliased select components
+                finalQuery.Append(selectStatementQueryText);
+
+                if (aliasedSelectStatements)
+                {
+                    finalQuery.Append($", ");
+                }
+            }
+
+            finalQuery.Append(selectStatementAliasedQueryText);
 
             // build the from string
-            finalQuery.Append(' ').Append(_fromQuery.GetQueryText());
+            finalQuery.Append(' ').Append(QueryConstants.From).Append(' ');
+            finalQuery.Append(_fromQuery.GetQueryText());
 
             // build the where string
-            finalQuery.Append(' ').Append(_whereStatement.GetQueryText());
+            string whereGetQueryText = _whereStatement.GetQueryText();
+
+            if (whereGetQueryText.Length > 0)
+            {
+                finalQuery.Append(' ').Append(QueryConstants.Where).Append(' ');
+            }
+
+            finalQuery.Append(_whereStatement.GetQueryText());
```

Comment:
shouldn't this be inside the previous if statement block?

Reply:
The code handles this so it's an empty string in the case of no where statements, so this would be a no-op, but I agree it might as well be moved in